sdk/keyvault/azure-security-keyvault-keys/src/keyvault_protocol.cpp

```diff
@@ -10,42 +10,12 @@
 using namespace Azure::Security::KeyVault;
 using namespace Azure::Core::Http::_internal;
 
-Azure::Core::Http::Request _detail::KeyVaultProtocolClient::CreateRequest(
-    Azure::Core::Http::HttpMethod method,
-    Azure::Core::IO::BodyStream* content,
```

Comment:
Aren't you diverging the protocol from other services? we should come up with a common protocol for keyvault before making changes.

Reply:
I am actually removing the protocol client here.  After reviewing it with Jeff, we won't use callbacks-model for the implementation as it is hard to read/understand.

Each client needs to use its own helper methods but should not use an extra class for wrapping the core pipeline which takes callbacks for serialize and de-serialize. 

The main motivation of this PR is to remove the callbacks and use an structured pattern for implementing each API